### PR TITLE
Introduce new API to BudgetAccountants.

### PR DESCRIPTION
**Pull request by zachferr**
## Description
This change covers step 1 of #78 

## Affected Dependencies
N/A

## How has this been tested?
- Ran tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: pipeline_dp/budget_accounting.py [hunk 1]
```diff
@@ -85,7 +91,9 @@ class BudgetAccountant(abc.ABC):
     def request_budget(self,
                        noise_kind: NoiseKind,
                        sensitivity: float = 1,
-                       weight: float = 1) -> MechanismSpec:
+                       weight: float = 1,
+                       count: int = 1,
+                       noise_standard_deviation: float = 1) -> MechanismSpec:
```
Comment (dvadym) [inline pipeline_dp/budget_accounting.py:96]: Nit: float -> Optional[float] = None (and in other places)

File: pipeline_dp/budget_accounting.py [hunk 2]
```diff
@@ -202,7 +214,9 @@ def __init__(self,
     def request_budget(self,
                        noise_kind: NoiseKind,
                        sensitivity: float = 1,
-                       weight: float = 1) -> MechanismSpec:
+                       weight: float = 1,
+                       count: int = 1,
```
Comment (dvadym) [inline pipeline_dp/budget_accounting.py:218]: Please raise NotImplementedError exception when count != 1 and noise_standard_deviation != None

File: pipeline_dp/budget_accounting.py [hunk 3]
```diff
@@ -116,7 +124,9 @@ def __init__(self, total_epsilon: float, total_delta: float):
     def request_budget(self,
                        noise_kind: NoiseKind,
                        sensitivity: float = 1,
-                       weight: float = 1) -> MechanismSpec:
+                       weight: float = 1,
```
Comment (dvadym) [inline pipeline_dp/budget_accounting.py:127]: Please raise NotImplementedError exception when count != 1 and noise_standard_deviation != None

**Review by dvadym — COMMENTED** [review]
A few minor comments

**Comment by dvadym** [comment]
Please pull new changes, [PR](https://github.com/OpenMined/PipelineDP/pull/71) with changes in this file has just been merged

File: pipeline_dp/budget_accounting.py (hunk 3 above)
Comment (zachferr) [reply pipeline_dp/budget_accounting.py]: Done

File: pipeline_dp/budget_accounting.py (hunk 2 above)
Comment (zachferr) [reply pipeline_dp/budget_accounting.py]: Done

File: pipeline_dp/budget_accounting.py (hunk 1 above)
Comment (zachferr) [reply pipeline_dp/budget_accounting.py]: Done

**Review by dvadym — APPROVED** [review]
Thanks!

**Pull request merged**